stan/math/rev/core/operator_equal.hpp

```diff
@@ -18,35 +19,50 @@ namespace math {
    \end{cases}
    \f]
  *
+ * @tparam Var1 value type of a var
+ * @tparam Var2 value type of a var
  * @param a First variable.
  * @param b Second variable.
  * @return True if the first variable's value is the same as the
  * second's.
  */
-inline bool operator==(const var& a, const var& b) {
+template <typename Var1, typename Var2, require_all_var_t<Var1, Var2>...>
```

Comment:
[optional]
This could be taken to prim and generalized with `value_of_rec` to arbitrary autodiff and scalar types in a single definition.   That assumes `value_of_rec(1.0)` is just `1.0`---if not, we should probalby do that with `value_of_rec`, too.

Same goes for `!=` and the other comparisons (`>, <, >=, <=`)

Reply:
I agree but let's flag this as an issue and do it in another PR once this goes through. 

On a side note, do we think of `fvar`'s as a "heavy" type? we may want specializations that forward for `fvar`'s with `var`s or `fvar`'s inside